### ISSUE TITLE
Add bookmarks to folders

### ISSUE THREAD

**Issue by mildred** [filebrowser/filebrowser#658]
**Is your feature request related to a problem? Please describe.**
I am using filebrowser to manage a Hugo static site, and the root directory is containing many things that might  frighten the non technical user. I need bookmarks in the sidebar to link to places people are more likely to go to.

**Describe the solution you'd like**
Bookmarks, a button is shown on a filder to bookmark it, it show then in the sidebar. Clicking again on the button removes the bookmark.

**Describe alternatives you've considered**
Static bookmarks from configuration file

**Additional context**
I already have an implementation

**Comment by hacdias** [comment]
This is an interesting idea. What do you think @1138-4EB?

**Comment by eine** [comment]
I think it is a good idea. Precisely, it is similar to filebrowser/frontend#55, which are *command bookmarks*. Some thoughts about it:

- How many lists of bookmarks will there be? A single one for all the users, one for each user, or one for each group of users?
- Who will be able to add/remove bookmarks? Only the admins? 
- If the list is per user/group, how can the admin see the list corresponding to each of them? I.e., does the field need to be added to the user settings in the frontend?
- Should the list(s) be modifiable through the CLI? Does it belong to subcommand `config` or `users`?

---

Moreover, I'd like to bring some additional thoughts about the frontend. I think that the sidebar takes too much space for the functionality it offers. ATM, only five buttons are shown (along with the version and a link to the help). I think that buttom `My files` can be removed, and use the 'home' symbol of the filebrowser logo instead. The others can be moved to the top bar. We can add separators in the top bar if that helps.

I'm bringing this now, because I think that 'Bookmarks' should be a button in the top bar, with a dropdown menu. The first item in the dropmenu would be to add the current location to the list. The other elements in the menu would be the items in the list (the bookmarks).

**Comment by mildred** [comment]
I was going for a list of bookmarks per user, and of course, each user could add or remove their bookmarks. Although mu use case involves global bookmarks, I don't believe that should how they should function. Bookmarks were always per-user, so I'm going for that in the implementation.

That being said, the per-user bookmarks are initialized from a the global configuration, but I'm not completely updated on how that global config works with the new version of filebrowser and how to change it.

> Moreover, I'd like to bring some additional thoughts about the frontend. I think that the sidebar takes too much space for the functionality it offers. ATM, only five buttons are shown (along with the version and a link to the help). I think that buttom My files can be removed, and use the 'home' symbol of the filebrowser logo instead. The others can be moved to the top bar. We can add separators in the top bar if that helps.

I agree the sidebar takes too much space currently for what it offers, but I disagree with putting everything on the top bar without distinction. Everything there is an icon, and when items are selected in a folder view, there are many more icons and it's hard to know what they all do.

Perhaps the multiple selection button could be completely removed and be replaced by a checkbox on each item on the folder view, a checkbox that can appear only after the first item has been selected.

Perhaps actions that are relative to the current selection could appear visually different, on a menu opened by a blue button with "N selected items" label. Also, contextual actions (like download, upload, information, bookmark) should be separated from non contextual actions (switch view, multiple selection, open terminal)

And I believe the sidebar is still nice for bookmarks and few other things as well, what about instead having a button to hide it when you don't want to see it?

Here is a mockup I made for it
![scan-browser-mockup](https://user-images.githubusercontent.com/33804/52204825-850cbd00-2875-11e9-819a-05b44425a111.png)



- the sidebar button left to the home button toggles the sidebar on and off
- the sidebar contains only bookmarks for now
- the toolbar is split
    - the top toolbar contains UI actions, on the left action relative to the folder view (switch view) and on the right, actions relative to the whole application (toggle terminal, and a menu button to settings and log-out)
    - the bottom bar is on the right of the breadcrumb contains slightly smaller buttons with actions relative to the current folder or the current selected item. The right part is about read-only actions, share button, ... On the left, we have a + button that opens up a menu that allows to add content (new empty file, new empty folder, upload file). Then, when a selection is made, a new menu appears with actions involving the selection (rename, copy, move, delete)

I plan to keep the icons on the menus, I just did not draw them
